### PR TITLE
 Support descriptors with checksums in initpegoutwallet 

### PR DESCRIPTION
**Pull request by stevenroose**
Non-pkh descriptor types will be re-enabled once entire liquid network functionary set is updated to use elements daemons.

### CODE REVIEW

**Comment by instagibbs** [comment]
utACK https://github.com/ElementsProject/elements/pull/726/commits/28dac17c4d6bbbbee28b76596a1baf042f2f3696

**Comment by instagibbs** [comment]
Turns out there's a bug because no test of the key origin-stripping code:

```
diff --git a/src/wallet/rpcwallet.cpp b/src/wallet/rpcwallet.cpp
index 25893c115..0d26b9081 100644
--- a/src/wallet/rpcwallet.cpp
+++ b/src/wallet/rpcwallet.cpp
@@ -5018,7 +5018,7 @@ UniValue initpegoutwallet(const JSONRPCRequest& request)

     // Strip off leading key origin
     if (xpub_str.find("]") != std::string::npos) {
-        xpub_str = xpub_str.substr(xpub_str.find("]"), std::string::npos);
+        xpub_str = xpub_str.substr(xpub_str.find("]")+1, std::string::npos);
     }

     // Strip off following range
```

in one of the non-checksummed examples you can add `[deadbeef/44h/0h/0h]` and make sure it works correctly(same as before).

File: test/functional/feature_pak.py [hunk 1]
```diff
@@ -63,6 +63,8 @@ def run_test(self):
 
         assert_raises_rpc_error(-8, "bip32_counter must be between 0 and 1,000,000,000, inclusive.", self.nodes[1].initpegoutwallet, xpub, 1000000001)
 
+        assert_raises_rpc_error(-8, "bitcoin_descriptor is not a valid descriptor string", self.nodes[1].initpegoutwallet, "[deadbeef/44h/0h/0h]")
```
Comment (instagibbs) [inline test/functional/feature_pak.py]: this error is pretty obscure(does this mean it's fixed? I can't really tell), can you instead insert it into one of the example descriptors that doesn't have it, and make sure it passes checks?

Comment (stevenroose) [reply test/functional/feature_pak.py]: hmm yeah sorry, this doesn't make sense

Comment (stevenroose) [reply test/functional/feature_pak.py]: fixed, btw

**Comment by instagibbs** [comment]
utACK https://github.com/ElementsProject/elements/pull/726/commits/55c4c6cd200cc8e6954c213fd117a29407dabe23

**Comment by stevenroose** [comment]
ACK 5af17a42bceb5d2b002c3848f58c8d35082c896b

**Pull request merged**